yt_dlp/extractor/dailymotion.py

```diff
@@ -232,6 +234,13 @@ def _extract_embed_urls(cls, url, webpage):
         for mobj in re.finditer(
                 r'(?s)DM\.player\([^,]+,\s*{.*?video[\'"]?\s*:\s*["\']?(?P<id>[0-9a-zA-Z]+).+?}\s*\);', webpage):
             yield from 'https://www.dailymotion.com/embed/video/' + mobj.group('id')
+        for mobj in re.finditer(
+                r'(?s)<script\s+?[^>]*?\bsrc=(["\'])((?:https?:)?//[^>]+\.dailymotion\.com/player/(?:(?!\1).)+)\1[^>]*?>', webpage):
+            attrs = extract_attributes(mobj.group(0))
+            player_url = url_or_none(attrs.get('src'))
+            video_id = attrs.get('data-video')
+            if all((player_url, video_id)):
+                yield f'{player_url.replace(".js", ".html")}?video={video_id}'
```

Comment:
There appear to be two types of embeds video and playlist embeds. Could you extend this PR to also match `data-playlist` embeds?
Also, please add `_WEBPAGE_TESTS`. Here two sample URLs:
- Video: https://www.financialounge.com/video/2024/08/01/borse-europee-in-rosso-dopo-la-fed-a-milano-volano-mediobanca-e-tim-edizione-del-1-agosto/
- Playlist: https://www.cycleworld.com/blogs/ask-kevin/ducati-continues-to-evolve-with-v4/